uniflow/op/extract/split/markdown_header_splitter.py

```diff
@@ -0,0 +1,143 @@
+"""Transform markdown op."""
+
+import copy
+from typing import Dict, List, Sequence, TypedDict
+
+from uniflow.node import Node
+from uniflow.op.op import Op
+
+
+class HeaderType(TypedDict):
+    """Header type as typed dict."""
+
+    level: int
+    name: str
+    data: str
+
+
+class LineType(TypedDict):
+    """Line type as typed dict."""
+
+    metadata: Dict[str, str]
+    content: str
```

Comment:
I would suggest to use `dataclass` to enforce object-oriented design instead of `TypeDict` which only provide a Json dict interface.